controllers/capabilities/methods.go

```diff
@@ -105,10 +105,18 @@ func (t *ProductThreescaleReconciler) createNewMethods(desiredNewMap map[string]
 
 func (t *ProductThreescaleReconciler) processNotDesiredMethods(notDesiredMap map[string]threescaleapi.MethodItem) error {
 	for _, notDesiredMethod := range notDesiredMap {
-		err := t.productEntity.DeleteMethod(notDesiredMethod.ID)
+		methodNotInUse, err := t.isMethodNotInUse(notDesiredMethod.ID)
 		if err != nil {
 			return err
 		}
+		if methodNotInUse {
```

Comment:
I don't think we need this Valery. 
WDYT if all we would do is to capture this specific error and make it more understandable?

Reply:
@MStokluska , I closed PR, as checked more, and agree with your notes in Task.
Regarding warning/error  to be more clear - also looks like it's clear enough what happens

```
$ oc describe product product2
..........
......
Status:
  Conditions:
    Last Transition Time:  2024-02-26T12:40:48Z
    Message:               Task failed SyncMethods: Error sync product methods, process not DesiredMethods [operatedproduct2]: product [operatedproduct2] delete method: error calling 3scale system - reason: {"errors":{"base":["Method is used by the latest gateway configuration and cannot be deleted"]}} - code: 403

```